### PR TITLE
Implement the BIR writer

### LINKED ISSUE #26147: Implement the BirWriter
**Description:**
The BirWriter should be re-implemented to work with the new Project API

### PR DESCRIPTION
**Pull request by azinneera**
Fixes #26147 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/ModuleCompilation.java [hunk 1]
```diff
@@ -95,6 +96,15 @@ private void compile() {
     public void getSemanticModel() {
         throw new UnsupportedOperationException();
     }
+
+    /**
+     * Write the BIR of the compiled module to the provided path.
+     *
+     * @param birFilePath BIR filepath
+     */
+    public void writeBir(Path birFilePath) {
```
Comment (azinneera) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/ModuleCompilation.java:105]: Change this to a utility method

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/ModuleContext.java [hunk 2]
```diff
@@ -193,4 +193,8 @@ void compile(CompilerContext compilerContext, PackageName packageName) {
         compilerPhaseRunner.compile(pkgNode);
         this.bLangPackage = pkgNode;
     }
+
+    BLangPackage bLangPackage() {
```
Comment (azinneera) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/ModuleContext.java:197]: Move the getter to ModuleCompilation

Comment (azinneera) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/ModuleContext.java:197]: Fixed in 00a2f6a

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/ModuleCompilation.java (hunk 1 above)
Comment (azinneera) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/ModuleCompilation.java]: Fixed in 00a2f6a

**Review by hevayo — APPROVED** [review]

**Pull request merged**